owasp-top10-2016-mobile/m5/panda_zap/mobile/lib/views/write_message.dart

```diff
@@ -0,0 +1,175 @@
+      height: 70.0,
+      child: Row(
+        children: <Widget>[
+          Expanded(
+            child: TextField(
+              textCapitalization: TextCapitalization.sentences,
+              controller: newMessageController,
+              onChanged: (value) {
+                messageTextToBeSent = value;
+              },
+              decoration: InputDecoration(hintText: "Type a message"),
+            ),
+          ),
+          IconButton(
+            icon: Icon(Icons.send),
+            iconSize: 25.0,
+            color: Theme.of(context).primaryColor,
+            onPressed: () async {
+              Message messageToBeSent = Message(Uuid().v4(), true, me.name,
+                  messageTextToBeSent, DateTime.now());
+              setState(
+                () {
+                  if (messageTextToBeSent.isNotEmpty) {
+                    widget.contactUser.messages.insert(0, messageToBeSent);
+                  }
+                },
+              );
+              await _sendMessageCaller(messageToBeSent, widget.contactUser);
+              newMessageController.clear();
+            },
+          ),
+        ],
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    List<Message> messages = widget.contactUser.messages;
+    int userIndex = widget.userIndex;
+    return Scaffold(
+      backgroundColor: Theme.of(context).primaryColor,
+      appBar: PreferredSize(
+        preferredSize: Size.fromHeight(70.0),
+        child: AppBar(
+          iconTheme: IconThemeData(
+            color: Colors.white,
+          ),
+          title: Text(
+            allUsers[userIndex].name,
+            style: TextStyle(
+              color: Colors.white,
+            ),
+          ),
+          centerTitle: true,
+          elevation: 0.0,
+        ),
+      ),
+      body: GestureDetector(
+        onTap: () => FocusScope.of(context).unfocus(),
+        child: Column(
+          children: <Widget>[
+            Expanded(
+              child: Container(
+                decoration: BoxDecoration(
+                  color: Colors.white,
+                  borderRadius: BorderRadius.only(
+                    topLeft: Radius.circular(30.0),
+                    topRight: Radius.circular(30.0),
+                  ),
+                ),
+                child: ClipRRect(
+                  borderRadius: BorderRadius.only(
+                    topLeft: Radius.circular(30.0),
+                    topRight: Radius.circular(30.0),
+                  ),
+                  child: ListView.builder(
+                    reverse: true,
+                    padding: EdgeInsets.only(top: 15.0),
+                    itemCount: messages.length,
+                    itemBuilder: (BuildContext context, int index) {
+                      // timer = Timer.periodic(
+                      //     Duration(seconds: 10), (Timer t) => setState(() {}));
+                      timer = Timer.periodic(Duration(seconds: 10),
```

Comment:
What is this timer? Can we remove the comment above?

Reply:
Nice catch! I'll remove those comments.

While in the `write_message` screen we need to be able to see new messages arriving. I used a timer to call the `setState()` method, which rebuilds the component and adds the new messages.